### PR TITLE
Move Extensions to use react-query instead of redux

### PR DESCRIPTION
**Pull request by briangleeson**
Remove the unused Extensions-related redux code.

# Changes

Move Extensions to use react-query instead of redux. Remove the unused Extensions-related redux code.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/main/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/main/standards.md#principles) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/main/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/dashboard/blob/main/CONTRIBUTING.md)
for more details._

### CODE REVIEW

File: src/containers/App/App.js [hunk 1]
```diff
@@ -184,10 +186,17 @@ export function App({
...
 
   const showLoadingState = isPropertiesPlaceholder || isMessagesPlaceholder;
-  const isFetchingConfig = isFetchingProperties || isFetchingMessages;
-  const loadingConfigError = propertiesError || messagesError;
+  const isFetchingConfig =
+    isFetchingProperties || isFetchingMessages || isFetchingExtensions;
+  const loadingConfigError =
+    propertiesError || messagesError || extensionsError;
+
+  function fetchData() {
```
Comment (briangleeson) [inline src/containers/App/App.js:194]: @AlanGreene wasnt 100% sure if this it the correct approach here for fetching the data via websocket

File: src/reducers/extensions.js [hunk 2]
```diff
@@ -1,78 +0,0 @@
...
-import keyBy from 'lodash.keyby';
-
-import {
-  createErrorMessageReducer,
-  createIsFetchingReducer
-} from './reducerCreators';
-
-const type = 'Extension';
-
-export function mapResourceExtension(extension) {
```
Comment (briangleeson) [inline src/reducers/extensions.js:24]: @AlanGreene is it safe to remove the `mapResourceExtension` and `byName` functions in this reducer?

**Comment by briangleeson** [comment]
There are some failures in the SideNav tests locally. Will need to figure out how to fix those

File: src/containers/App/App.js (hunk 1 above)
Comment (AlanGreene) [reply src/containers/App/App.js:194]: yes this is good for now, after all resources have been fully migrated to react-query we'll be centralising this using [`invalidateQueries`](https://react-query.tanstack.com/reference/QueryClient#queryclientinvalidatequeries) and letting react-query take care of refetching all active queries automatically.

File: src/containers/App/App.js [hunk 3]
```diff
@@ -184,10 +186,17 @@ export function App({
   });
 
   const showLoadingState = isPropertiesPlaceholder || isMessagesPlaceholder;
-  const isFetchingConfig = isFetchingProperties || isFetchingMessages;
-  const loadingConfigError = propertiesError || messagesError;
+  const isFetchingConfig =
+    isFetchingProperties || isFetchingMessages || isFetchingExtensions;
```
Comment (AlanGreene) [inline src/containers/App/App.js:190]: no need to include extensions in this, it's only for holding off on setting global config such as the selected namespace which may depend on deployment settings (e.g. single namespace mode)

File: src/containers/SideNav/SideNav.js [hunk 4]
```diff
@@ -52,6 +63,10 @@ function SideNav(props) {
 
   const { namespace } = match?.params || {};
 
+  const { data: extensions = [], refetch } = useExtensions();
+
+  useWebSocketReconnected(refetch, webSocketConnected);
```
Comment (AlanGreene) [inline src/containers/SideNav/SideNav.js:68]: no need for this here since it's already handled in App

File: src/reducers/extensions.js (hunk 2 above)
Comment (AlanGreene) [reply src/reducers/extensions.js:24]: Once the logic in src/containers/Extensions/Extension.js that was relying on the `extensionType` is updated it should be safe to remove 👍

File: src/containers/SideNav/SideNav.js (hunk 4 above)
Comment (briangleeson) [reply src/containers/SideNav/SideNav.js:68]: Ok, so `extensions` should be passed in as a prop to SideNav from App?

Comment (AlanGreene) [reply src/containers/SideNav/SideNav.js:68]: sorry just meant the websocket piece, they're already being refetched on reconnect in App, so we can just call the `useExtensions` hook here and will benefit from the caching etc. already being done by react-query

File: src/api/utils.js [hunk 5]
```diff
@@ -186,8 +186,8 @@ export function useWebSocket({ kind: _ }) {
   }, [queryClient, webSocket]);
 }
 
-export function useCollection(kind, api, params) {
-  const query = useQuery([kind, params], () => api(params));
+export function useCollection(kind, api, params, queryConfig) {
+  const query = useQuery([kind, params], () => api(params, queryConfig));
```
Comment (AlanGreene) [inline src/api/utils.js:190]:
```suggestion
  const query = useQuery([kind, params], () => api(params), queryConfig);
```

`queryConfig` should be passed as the third param to `useQuery`

File: src/containers/App/App.js [hunk 6]
```diff
@@ -185,9 +182,25 @@ export function App({
...
+    data: extensions = [],
+    error: extensionsError,
+    refetch: refetchExtensions
+  } = useExtensions(
+    { namespace: tenantNamespace || ALL_NAMESPACES },
+    { enabled: !isFetchingConfig }
+  );
+
+  const loadingConfigError =
+    propertiesError || messagesError || extensionsError;
```
Comment (AlanGreene) [inline src/containers/App/App.js:196]: no need to include extensions here as they don't block / affect the rest of the app, if we fail to load them we just won't show the nav items for them but everything else in the app works as expected.

File: src/containers/Extensions/Extensions.js [hunk 7]
```diff
@@ -19,28 +19,31 @@ import {
...
   useTitleSync({
     page: intl.formatMessage({
       id: 'dashboard.extensions.title',
       defaultMessage: 'Extensions'
     })
   });
 
+  const { data: extensions = [], error, isFetching, refetch } = useExtensions();
+
+  useWebSocketReconnected(refetch, webSocketConnected);
```
Comment (AlanGreene) [inline src/containers/Extensions/Extensions.js:45]: no need for the refetch here since it's handled centrally in App.js `fetchData`

**Review by AlanGreene — CHANGES_REQUESTED** [review]
Thanks @briangleeson. Just a few small changes then it's ready to go.

**Review by AlanGreene — APPROVED** [review]
Looks good, thanks @briangleeson 
/meow

**Pull request merged**